pkg/limayaml/defaults.go

```diff
@@ -905,11 +907,97 @@ func NewVMType(driver string) VMType {
 	}
 }
 
-func ResolveVMType(s *string) VMType {
-	if s == nil || *s == "" || *s == "default" {
+func isExistingInstanceDir(dir string) bool {
+	// existence of "lima.yaml" does not signify existence of the instance,
+	// because the file is created during the initialization of the instance.
+	for _, f := range []string{
+		filenames.HostAgentStdoutLog, filenames.HostAgentStderrLog,
+		filenames.VzIdentifier, filenames.BaseDisk, filenames.DiffDisk, filenames.CIDataISO,
+	} {
+		file := filepath.Join(dir, f)
+		if _, err := os.Lstat(file); !errors.Is(err, os.ErrNotExist) {
+			return true
+		}
+	}
+	return false
+}
+
+func ResolveVMType(y, d, o *LimaYAML, filePath string) VMType {
+	// Check if the VMType is explicitly specified
+	for i, f := range []*LimaYAML{o, y, d} {
+		if f.VMType != nil && *f.VMType != "" && *f.VMType != "default" {
+			logrus.Debugf("ResolveVMType: resolved VMType %q (explicitly specified in []*LimaYAML{o,y,d}[%d])", *f.VMType, i)
+			return NewVMType(*f.VMType)
+		}
+	}
+
+	// If this is an existing instance, guess the VMType from the contents of the instance directory.
+	// Note that the instance directory may be created by a previous version of Lima.
```

Comment:
Added a comment to clarify that vz-identifier exists since v0.14.
Let me know if more comments should be added.

Reply:
> Let me know if more comments should be added.

To me this part was confusing: "the instance directory may be created by a previous version of Lima."

I thought of versions **before** 1.0 as "previous versions". But in that case the `vzIdentifier` file cannot exist because the `vmType` was always defaulting to `qemu`.

So the following block is really for **future** versions of Lima (post-1.0), where the instance directory may have been created by 1.0 or later. In which case the `vmType` could be empty, but the `vzIdentifier` file will exist.

I actually think the code would be clearer when we delete the 2nd line of the comment, which right now is just misleading (at least to me).

Reply:
> I actually think the code would be clearer when we delete the 2nd line of the comment, which right now is just misleading (at least to me).

done